torchdata/dataloader2/dataloader2.py

```diff
@@ -88,25 +121,16 @@ def __iter__(self) -> Iterator[T_co]:
                 self.reading_service.initialize_iteration()
 
             self._datapipe_iter = iter(self.datapipe)
```

Comment:
There is a problem that `self._datapipe_iter` is never updated when we want to have new `_DataLoader2Iterator`

Reply:
I think it is? When `__iter__` of `DataLoader2` is called, it first checks if `self._reset_iter == True` (an indicator whether an active iterator has been read at least once). If that is `True`, `self._datapipe_iter` gets updated and `self._reset_iter` is set to `False`. 

Let's discuss offline to make sure I know what you mean.